src/lib/ptree/ptree.lua

```diff
@@ -225,7 +225,7 @@ end
 function Manager:handle_notification_peer(peer)
    local q = queue.new()
    self.notification_peers[q] = true
-   function q:close()
+   function q.close()
```

Comment:
This change is not needed. I explain why.

In Lua when an object is called using ':' the caller object is passed as an implicit parameter (that makes possible to reference `self` inside the callee function). The method definition can either use ':' or '.'. If using ':' there's nothing else to do, all good. If the method definition uses '.', then it generally means it doesn't expect any arguments, but it could. Example:

```lua
local t = {}

function t:foo(arg)
   if self then print('self is implicit') end
   print('foo: '..arg)
end
function t.bar(caller, arg)
   if not self then print("self doesn't exist") end
   if caller then print("caller exists") end
   print('bar: '..arg)
end

print(t:foo(42))
print(t:bar(42))
```
Whether to use ':' or '.' in an object method definition which doesn't have arguments depend very much on each person coding style. I generally prefer to use ':' in case the method is extended with arguments in the future. But as for "class methods", such as `new` I prefer to use '.'.

Reply:
Yeah so using `:` means `self` is passed through implicitly - rather than explicitly. 

The problem with using `:` on `q:close()` is that `self.notification_peers` (where `self` is `Manager`)  is referenced *inside* the `close()` function. At that point, if using `:`, `self` is `queue` rather than `Manager` and does not have a `notification_peers` var.

Since the `close` method on `q` doesn't actually refer to itself in any manner, it should be safe to use `q.close` which leaves `self` referencing `Manager`.

Reply:
OK, makes perfect sense to me. Since the diff didn't show the body of the function I wrongly assumed it was aesthetic change. So I think this is the fix for the `attempt to index field 'notification_peers'` you mentioned in Slack.

Reply:
Yep, should've mentioned that 😃 Fixes the index issue.